backend/gn_plugin_depobio/blueprint.py

```diff
@@ -0,0 +1,97 @@
+from time import sleep
+
+from flask import Blueprint, abort
+
+import logging
+from geonature.core.gn_meta.models import TAcquisitionFramework
+from geonature.utils.env import db
+from geonature.utils.errors import GeoNatureError
+from geonature.core.gn_permissions import decorators as permissions
+from utils_flask_sqla.response import json_resp
+from gql.transport.exceptions import TransportQueryError
+from werkzeug.exceptions import Forbidden, NotFound
+from .demarches_simplifiees import DemarchesSimplifieesConnexion, ErrorCode
+from .mail_builder import MailBuilder
+
+log = logging.getLogger()
+blueprint = Blueprint("plugin_depobio", __name__)
+
+
+@blueprint.route("/extended_af_publish/<int:af_id>", endpoint="extended_af_publish")
+@permissions.check_cruved_scope("E", module_code="METADATA")
+@json_resp
+def publish_acquisition_framework_mail(af_id):
+    """
+    Method for sending a mail during the publication process
+    Parameters
+    ----------
+    af_id Identifiant of acquisition framework
+
+    Returns Mail sent
+    -------
+
+    """
+    acquisition_framework = db.session.get(TAcquisitionFramework, af_id)
+    mail_builder = MailBuilder(acquisition_framework)
+    try:
+        mail_builder.send_mail()
+    except GeoNatureError as error:
+        log.error(str(error))
+        raise GeoNatureError(error)
+    return mail_builder.mail
+
+
+def convert_error_to_exception(error: TransportQueryError, folder_number: int) -> Exception:
+    error_code = error.errors[0]["extensions"]["code"]
+    if error_code == ErrorCode.NOT_FOUND:
+        result = NotFound(f"Le dossier numéro {folder_number} n'existe pas")
+        result.printable = True
+    elif error_code == ErrorCode.FORBIDDEN:
+        result = Forbidden(
+            f"Le dossier numéro {folder_number} ne peux pas être visualisé. Vérifiez que votre numéro de "
+            f"dossier appartient à la bonne démarche"
```

Comment:
```suggestion
            f"Le dossier numéro {folder_number} ne peut pas être récupéré. Vérifiez que votre numéro de "
            "dossier appartient à la bonne démarche"
```

- "peut" au lieu de "peux"
- Retrait d'une f-string non nécessaire
- "récupéré" plutôt que "visualisé" : plus générique, et valable pour les différents cas d'utilisation / peut-être une 3ème option meilleure encore, mais je ne trouve pas